pkg/services/oracle/network_test.go

```diff
@@ -16,3 +20,45 @@ func TestIsReserved(t *testing.T) {
 
 	require.False(t, isReserved(net.IPv4(8, 8, 8, 8)))
 }
+
+func TestDefaultClient_RestrictedRedirectErr(t *testing.T) {
+	cfg := config.OracleConfiguration{
+		AllowPrivateHost: false,
+		RequestTimeout:   time.Second,
+	}
+	cl := getDefaultClient(cfg)
+
+	// Firstly, check URLs without redirection.
+	type testCase struct {
+		url        string
+		shouldFail bool
+	}
+	testCases := []testCase{
+		{"http://localhost:8080", true},
+		{"http://localhost", true},
+		{"https://localhost:443", true},
+		{"https://" + net.IPv4zero.String(), true},
+		{"https://" + net.IPv4(10, 0, 0, 1).String(), true},
+		{"https://" + net.IPv4(192, 168, 0, 1).String(), true},
+		{"https://[" + net.IPv6interfacelocalallnodes.String() + "]", true},
+		{"https://[" + net.IPv6loopback.String() + "]", true},
+
+		{"https://google.com", false}, // hope, Google won't try to steel our data and redirect to non-unicast.
+		{"https://" + net.IPv4(8, 8, 8, 8).String(), false},
```

Comment:
This creates a networking dependency in tests, not a good idea. Tests must not go outside for anything.

Reply:
But I have a special error handler for this:
```
if err != nil { // URL may be not available
	require.False(t, errors.Is(err, ErrRestrictedRedirect), err)
	require.False(t, strings.Contains(err.Error(), "IP is not global unicast"), err)
}
```
Otherwice we don't have non-error cases for this test, is it OK?

Reply:
It either should be faked in some way or not tested, but accessing public networks from a test is a problem. It will break `make test` on non-networked laptop, it will break in access fails for some strange reason.